README.md

```diff
@@ -1,2 +1,86 @@
-# nf-synstage
-Nextflow pipeline for staging files from Synapse
+# nf-synstage: Stage Synapse Files
+
+## Purpose
+
+The purpose of this Nextflow workflow is to automate the process of staging Synapse files in an accessible location (_e.g._ an S3 bucket). In turn, these staged files can be used as input for a general-purpose (_e.g._ nf-core) workflow that doesn't contain Synapse-specific steps for downloading data. This workflow is intended to be run first in preparation for other data processing workflows.
+
+Briefly, `nf-synstage` achieves this automation as follows:
+
+1. Extract all Synapse URIs (_e.g._ `syn://syn28521174`) from a given text file
+2. Download the corresponding files from Synapse in parallel
+3. Replace the Synapse URIs in the text file with their staged locations
+4. Output the updated text file so it can serve as input for another workflow
+
+## Quickstart
+
+The examples below demonstrate how you would stage Synapse files in an S3 bucket called `example-bucket`, but they can be adapted for other storage backends.
+
+1. Prepare your input text file containing the Synapse URIs. For example, the following CSV file follows the format required for running the [`nf-core/rnaseq`](https://nf-co.re/rnaseq/latest/usage) workflow.
+
+    **Example:** Uploaded to `s3://example-bucket/input.csv`
+
+    ```text
+    sample,fastq_1,fastq_2,strandedness
+    foobar,syn://syn28521174,syn://syn28521175,unstranded
+    ```
+
+2. Prepare your Synapse configuration file to authenticate the workflow. For more details, check out the [Authentication](#authentication) section.
+
+    **Example:** Uploaded to `s3://example-bucket/synapse_config.ini`
```

Comment:
I like this.  I feel like we could rename the ./synapseConfig to `.synapseConfig.ini`

Reply:
It makes it easier to identify it as a file in S3. Otherwise, it can look like an S3 "folder".